src/Device/Driver/AirControlDisplay.cpp

```diff
@@ -68,8 +69,27 @@ ParsePAAVS(NMEAInputLine &line, NMEAInfo &info)
     unsigned volume;
     if (line.ReadChecked(volume))
       info.settings.ProvideVolume(volume, info.clock);
+  } else if (type == "XPDR"sv) {
+    /*
+    $PAAVS,XPDR,<SQUAWK>,<ACTIVE>,<ALTINH>
+    <SQUAWK> Squawk code value;
+             Octal unsigned integer value between 0000 and 7777 (digits 0–7).
+    <ACTIVE> Active flag;
+             0: standby (transponder is switched off / "SBY" mode)
+             1: active (transponder is switched on / "ALT" or "ON" mode
+                dependent of ALTINH)
+    <ALTINH> Altitude inhibit flag;
+             0: transmit altitude ("ALT" mode if active)
+             1: do not transmit altitude ("ON" mode if active)
+     */
+    unsigned transponder_code;
+    if (line.ReadChecked(transponder_code)) {
+      TCHAR buffer[12];
+      snprintf(buffer, sizeof(buffer), "%04u", transponder_code);
+      info.settings.transponder_code = TransponderCode::Parse(buffer);
+      info.settings.has_transponder_code.Update(info.clock);
```

Comment:
Here, you should check whether the parsed value was valid - XCSoar must never misbehave even in the presence of arbitrary malformed input.
In the past, you could easily crash XCSoar by feeding malformed NMEA with broken GPS coordinates - that's a critical bug.

Reply:
done (and error caused by TCHAR usage on WIN64 fixed).